src/types/nbodysystem.rs

```diff
@@ -98,4 +123,106 @@ impl NBodySystem {
                 },
             )
     }
+
+    pub fn set_output(&mut self, output: String) {
+        self.m_output = output;
+    }
+
+    pub fn set_limit(&mut self, limit: u16) {
+        self.m_limit = limit;
+    }
+
+    pub fn set_file(&mut self, file: bool) {
+        self.m_file = file;
+    }
+
+    fn output(&self) {
+        static OUTPUT: AtomicU16 = AtomicU16::new(1);
+
+        let count = OUTPUT.fetch_add(1, Ordering::SeqCst);
+
+        if count == self.m_limit {
+            if self.m_output == "plain_text" {
+                let output = self.output_plain_text();
+                if self.m_file {
+                    let _ = std::fs::write("output.txt", output);
+                } else {
+                    println!("{output}");
+                }
+            } else if self.m_output == "csv" {
+                self.output_csv().expect("Error writing to CSV file");
+            } else if self.m_output == "json" {
+                self.output_json().expect("Error writing to JSON file");
+            } else {
+                println!("Unknown output format");
+            }
+
+            OUTPUT.store(0, Ordering::SeqCst);
+        }
+    }
+
+    fn output_plain_text(&self) -> String {
+        let mut output = String::new();
+        for particle in self.m_particles.iter() {
```

Comment:
Remark there about SOLID violation regarding the nbodysystem class responsibilities. It's better to introduce a notification that __something__ has changed within the system, so the external Observer can dump that into into CSV/TXT/JSON/whatever format.

Reply:
Done